repos/system_upgrade/el7toel8/libraries/isccfg.py

```diff
@@ -0,0 +1,959 @@
+        """ Create iterator
+        :param comments: Include comments and whitespaces
+        :param start: Index for starting, None means beginning of section
+        """
+        self.parser = parser
+        self.section = section
+        self.current = None
+        self.key_wanted = True
+        self.comments = comments
+        self.waiting = None
+        if start is None:
+            start = section.start
+        self.start = start
+
+    def __iter__(self):
+        self.current = None
+        self.key_wanted = True
+        self.waiting = None
+        return self
+
+    def __next__(self):
+        index = self.start
+        cfg = self.section.config
+        if self.waiting:
+            self.current = self.waiting
+            self.waiting = None
+            return self.current
+        if self.current is not None:
+            index = self.current.end+1
+        if self.key_wanted:
+            val = self.parser.find_next_key(cfg, index, self.section.end)
+            self.key_wanted = False
+        else:
+            val = self.parser.find_next_val(cfg, None, index, self.section.end, end_report=True)
+            if val is not None and val.value() in self.parser.CHAR_DELIM:
+                self.key_wanted = True
+        if val is None:
+            if self.current is not None and self.current.end < self.section.end and self.comments:
+                self.current = ConfigSection(self.section.config, None,
+                                             index, self.section.end, ConfigSection.TYPE_IGNORED)
+                return self.current
+            raise StopIteration
+        if index != val.start and self.comments:
+            # Include comments and spaces as ignored section
+            self.waiting = val
+            val = ConfigSection(val.config, None, index, val.start-1, ConfigSection.TYPE_IGNORED)
+
+        self.current = val
+        return val
+
+    def next(self):
+        return self.__next__()
+
+
+class IscVarIterator(object):
+    """ Iterator for walking over parsed configuration
+
+        Creates sequence of ConfigVariableSection objects for a given
+        file or section.
+    """
+
+    def __init__(self, parser, section, comments=False, start=None):
+        """ Create iterator """
+        self.parser = parser
+        self.section = section
+        self.iter = IscIterator(parser, section, comments, start)
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        vl = []
+        try:
+            statement = next(self.iter)
+            while statement:
+                vl.append(statement)
+                if self.parser.is_terminal(statement):
+                    return ConfigVariableSection(vl, None, parent=self.section)
+                statement = next(self.iter)
+        except StopIteration:
+            if vl:
+                return ConfigVariableSection(vl, None, parent=self.section)
+        raise StopIteration
```

Comment:
```suggestion
                return ConfigVariableSection(vl, None, parent=self.section)
            raise
```

Reply:
would this change raise always StopIteration exception? I want to ignore StopIteration "errors", but not any other. It seems okay to me

Reply:
you're right